From 1ce4dc224418031afcdd7c1a3ab8f3451ea4d972 Mon Sep 17 00:00:00 2001
From: Pavel Grunt <pgrunt@redhat.com>
Date: Sun, 5 Apr 2015 15:45:00 +0200
Subject: [PATCH] virt-viewer-window: Set zoom when display is enabled and
 ready.

The display has default dimensions (100x100) when it is disabled.
Calculating the minimal zoom for the display will give wrong value
for the newly opened display.

It is better to wait for setting the zoom level to the moment when
the display is enabled and ready.

Related to: https://bugzilla.redhat.com/show_bug.cgi?id=1206460

(cherry picked from commit 998b13c881480ffea98dd8b65e0495c49fc31d2d)
---
 src/virt-viewer-window.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/src/virt-viewer-window.c b/src/virt-viewer-window.c
index c808a4c..f185eb9 100644
--- a/src/virt-viewer-window.c
+++ b/src/virt-viewer-window.c
@@ -1272,6 +1272,10 @@ display_show_hint(VirtViewerDisplay *display,
 
     hint = (hint & VIRT_VIEWER_DISPLAY_SHOW_HINT_READY);
 
+    if (hint && virt_viewer_display_get_enabled(display)) {
+        virt_viewer_window_set_zoom_level(self, self->priv->zoomlevel);
+    }
+
     gtk_widget_set_sensitive(GTK_WIDGET(gtk_builder_get_object(self->priv->builder, "menu-file-screenshot")), hint);
 }
 static gboolean
@@ -1301,7 +1305,6 @@ virt_viewer_window_set_display(VirtViewerWindow *self, VirtViewerDisplay *displa
     if (display != NULL) {
         priv->display = g_object_ref(display);
 
-        virt_viewer_window_set_zoom_level(self, priv->zoomlevel);
         virt_viewer_display_set_monitor(VIRT_VIEWER_DISPLAY(priv->display), priv->fullscreen_monitor);
         virt_viewer_display_set_fullscreen(VIRT_VIEWER_DISPLAY(priv->display), priv->fullscreen);
 
